### PR TITLE
Add force: true when using get_url module to better support upgrade

### PR DESCRIPTION
**Pull request by erikgb**
# Description

I was originally trying to fix an issue related to check mode when trying to upgrade Prometheus JMX Exporter. Somehow I didn't understand why some of the tasks were skipped in check mode - until I discovered https://github.com/ansible/ansible/issues/65687.

Then I actually tried to upgrade, and discovered that files were actually NOT updated. After some digging, it seems like we need to set the [force parameter](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/get_url_module.html#parameter-force) when using the `get_url` module, because the destination file does not change when modifying the version inventory variable. So without this fix, you will be stuck with the version installed initially.

Setting `force` to `true` is not recommended, since the file will be downloaded each time, but I think it should be ok in this case. Both the Prometheus JMX Exporter and Jolokia jar-files are relatively small....

UPDATE: Since reviewer did not want to change the actual (IMO buggy) behavior, I have now added two new variables that retains the existing behavoir (`force: false`), but allows users to force re-download of Jolokia and JMX Exporter to upgrade/downgrade the jar-files, thus making actual state equal to desired state.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

I have tested the change on actual host in our dev cluster, and also in check mode.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

**Comment by erikgb** [comment]
> ERROR: We have disabled builds for pull requests created by the open source communitydue to build system maintenance.We are working to finish it soon.

@JumaX @domenicbove IMO this is contributor unfriendly, and it has been like that for a while now. Please fix this ASAP - one way or the other.

**Comment by JumaX** [comment]
Have kicked off the build for this.

**Comment by erikgb** [comment]
@JumaX @nsharma-git Friendly ping on this one. Any chance to get it reviewed and merged?

File: roles/confluent.common/tasks/main.yml [hunk 1]
```diff
@@ -68,12 +68,12 @@
   when:
     - jolokia_enabled|bool
     - not jolokia_url_remote|bool
-    - not ansible_check_mode
 
 - name: Download Jolokia Jar
   get_url:
     url: "{{ jolokia_jar_url }}"
     dest: "{{ jolokia_jar_path }}"
+    force: true # Dest does not include version, so need to download and check if changed
```
Comment (nsharma-git) [inline roles/confluent.common/tasks/main.yml:76]: Can we make this a boolean parameter defaulting to false.

Comment (erikgb) [reply roles/confluent.common/tasks/main.yml:76]: > Can we make this a boolean parameter defaulting to false.

Yes, I can do that.

Comment (erikgb) [reply roles/confluent.common/tasks/main.yml:76]: I am having a hard time to find proper names for these eventual new variables. And I do not understand why we should have a variable to control if we should try to make actual state equal desired state? As it is now, there is no way to upgrade Prometheus or Jolokia. @nsharma-git can you please explain why you think a variable is better?

Comment (nsharma-git) [reply roles/confluent.common/tasks/main.yml:76]: IMO actual state and desired state may not be always same. We may not always want to get the latest one if there is some issue with it. Having controlled way to get the latest one would be always better.

Comment (erikgb) [reply roles/confluent.common/tasks/main.yml:76]: > We may not always want to get the latest one if there is some issue with it.

I definitely agree on that, but the issue here is that even if you change the version of Jolokia or JMX Exporter, your system will not be updated! And that is very confusing IMO. The reason being that the `get_url` module does not even attempt to do anything if the `dest` file is present. Another option would be to include the Jolokia/JMX-exporter version in the `dest` filename.

Comment (nsharma-git) [reply roles/confluent.common/tasks/main.yml:76]: We need to have the force flag, I'm just trying to not make it true as default value. 
We can also add some checks if file was updated when the force flag was set to true explicitly.

Comment (erikgb) [reply roles/confluent.common/tasks/main.yml]: @nsharma-git I have now added two new variables as requests. PTAL! 😊

Comment (nsharma-git) [reply roles/confluent.common/tasks/main.yml]: Thanks @erikgb 😊

**Pull request merged**